cmake/Platform/Mac/InstallUtils_mac.cmake.in

```diff
@@ -0,0 +1,168 @@
+        PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE GROUP_EXECUTE WORLD_READ WORLD_EXECUTE 
+    )
+
+endfunction()
+
+function(codesign_file file entitlement_file)
+
+    if (NOT @LY_ENABLE_HARDENED_RUNTIME@)
+        return()
+    endif()
+
+    if(EXISTS ${entitlement_file})
+
+        execute_process(COMMAND "/usr/bin/codesign" "--force" "--sign" "@LY_CODE_SIGN_IDENTITY@" "--deep" "-o" "runtime" "--timestamp" "--entitlements" "${entitlement_file}" "${file}"
+            TIMEOUT 300
+            OUTPUT_VARIABLE codesign_out
+            RESULT_VARIABLE codesign_ret
+        )
+    else()
+        execute_process(COMMAND "/usr/bin/codesign" "--force" "--sign" "@LY_CODE_SIGN_IDENTITY@" "--deep" "-o" "runtime" "--timestamp" "${file}"
+            TIMEOUT 300
+            OUTPUT_VARIABLE codesign_out
+            RESULT_VARIABLE codesign_ret
+        )
+    endif()
+
+    if(NOT ${codesign_ret} EQUAL "0")
+        message(FATAL_ERROR "Codesign operation for ${file_path} returned ${codesign_ret} with message ${codesign_out}")
+    endif()
+
+endfunction()
+
+function(codesign_python_framework_binaries framework_path)
+
+    if (NOT @LY_ENABLE_HARDENED_RUNTIME@)
+        return()
+    endif()
+
+    # The codesign "--deep" flag will only codesign binaries in folders with specific names.
+    # We need to codesign all the binaries that the "--deep" flag will miss.
+    file(GLOB_RECURSE files
+        LIST_DIRECTORIES false
+        "${framework_path}/Versions/3.7/bin/**"
+        "${framework_path}/Versions/3.7/lib/**"
+        "${framework_path}/Versions/3.7/Resources/**")
+
+    foreach(file ${files})
+        if(NOT EXISTS ${file})
+            file(REMOVE ${file})
+            continue()
+        endif()
+        cmake_path(SET path_var "${file}")
+        cmake_path(GET path_var EXTENSION LAST_ONLY extension)
+        set(should_codesign FALSE)
+        set(extension_skip_list ".dylib" ".so" ".7m")
+        if (NOT extension)
+            set(should_codesign TRUE)
+        elseif(extension IN_LIST extension_skip_list)
+            set(should_codesign TRUE)
+        endif()
+        if(${should_codesign})
+            codesign_file("${file}" "@LY_ROOT_FOLDER@/python/Platform/Mac/PythonEntitlements.plist")
+        endif()
+    endforeach()
+
+endfunction()
+
+function(ly_copy source_file target_directory)
+    
+    if("${source_file}" MATCHES "\\.[Ff]ramework[^\\.]")
+
+        # fixup origin to copy the whole Framework folder
+        string(REGEX REPLACE "(.*\\.[Ff]ramework).*" "\\1" source_file "${source_file}")
+
+    endif()
+    get_filename_component(target_filename "${source_file}" NAME)
+    file(COPY "${source_file}" DESTINATION "${target_directory}" FILE_PERMISSIONS @LY_COPY_PERMISSIONS@ FOLLOW_SYMLINK_CHAIN)
+
+    # Our Qt and Python frameworks aren't in the correct bundle format to be codesigned.
+    if("${target_filename}" MATCHES "(Qt[^\\.]+)\\.[Ff]ramework")
```

Comment:
Inside of square brackets the <dot> character isn't treated specially.
So this regex can be simplified to
```suggestion
    if("${target_filename}" MATCHES "(Qt[^.]+)\\.[Ff]ramework")
```

Reply:
Changing.

Reply:
Fixed